metadata/memory_lookup.go

```diff
@@ -0,0 +1,333 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/featureform/fferr"
+	"github.com/featureform/logging"
+	pb "github.com/featureform/metadata/proto"
+	"github.com/featureform/storage"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+// Create Resource Lookup Using ETCD
+type MemoryResourceLookup struct {
+	Connection storage.MetadataStorage
+}
+
+// Wrapper around Resource/Job messages. Allows top level storage for info about saved value
+type MemoryRow struct {
+	ResourceType ResourceType //Resource Type. For use when getting stored keys
+	//ResourceType string
+	StorageType StorageType //Type of storage. Resource or Job
+	Message     []byte      //Contents to be stored
+}
+
+type MemoryRowTemp struct {
+	//ResourceType ResourceType //Resource Type. For use when getting stored keys
+	ResourceType ResourceType
+	StorageType  StorageType //Type of storage. Resource or Job
+	Message      []byte      //Contents to be stored
+}
+
+// Takes a populated ETCD storage struct and a resource
+// Checks to make sure the given ETCD storage Object contains a Resource, not job
+// Deserializes Resource value into the provided Resource object
+func (s MemoryResourceLookup) ParseResource(res EtcdRow, resType Resource) (Resource, error) {
+	if res.StorageType != RESOURCE {
+		return nil, fmt.Errorf("payload is not resource type")
+	}
+
+	if !resType.Proto().ProtoReflect().IsValid() {
+		return nil, fmt.Errorf("cannot parse to invalid resource")
+	}
+
+	if res.Message == nil {
+		return nil, fmt.Errorf("cannot parse invalid message")
+	}
+
+	if err := proto.Unmarshal(res.Message, resType.Proto()); err != nil {
+		return nil, err
+	}
+
+	return resType, nil
+}
+
+// Returns an empty Resource Object of the given type to unmarshal etcd value into
+func (lookup MemoryResourceLookup) createEmptyResource(t ResourceType) (Resource, error) {
+	var resource Resource
+	switch t {
+	case FEATURE:
+		resource = &featureResource{&pb.Feature{}}
```

Comment:
is there a reason why we don't just return? 
```
    switch t {
    case FEATURE:
        return &featureResource{&pb.Feature{}}, nil

Reply:
This should be a top level function somewhere else